components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -91,6 +95,27 @@ public static String getSCIMGroupURL() {
     }
 
     private static String getSCIMURL() {
+
+        String scimURL;
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            try {
+                scimURL = ServiceURLBuilder.create().addPath(SCIMCommonConstants.SCIM2_ENDPOINT).build()
+                        .getAbsolutePublicURL();
+            } catch (URLBuilderException e) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Error occurred while building the SCIM2 endpoint with tenant " +
+                            "qualified URL.", e);
+                }
+                // Fallback to legacy way in error scenarios.
+                scimURL = getSCIMURLLegacy();
+            }
+        } else {
+            scimURL = getSCIMURLLegacy();
+        }
+        return scimURL;
+    }
+
+    private static String getSCIMURLLegacy() {
```

Comment:
We don't need specifically check the `IdentityTenantUtil.isTenantQualifiedUrlsEnabled()` here. It's already checked inside the ServiceURLBuilder as well. So we can just use it and completely get rid of legacy method as well.